### PR TITLE
Support prisma-language-server

### PR DESCRIPTION
**Pull request by LumaKernel**
~WIP~

- [x] Installer for `prisma-lanaguage-server`
- [x] Installer for `prisma-fmt`
  - There're 3 ways,
    - from `npm i prisma` and use `node_modules/.bin/prisma fmt`,
    - from `npm i @prisma/engines` and use `node_modules/@prisma/engines/prisma-fmt-{PLATFORM_SPECIFIC}`,
    - and `git clone https://github.com/prisma/prisma-engines && cd prisma-engines && cargo build --release prisma-fmt`, and use `prisma-fmt`
  - Using `prisma fmt` looks smart (internally using second one) , but `prisma-language-server` does not support that. ( also `npx prisma fmt` ) . I'll issue for that...
  - MEMO: Binaries come from https://binaries.prisma.sh
- [x] Check installation and working with Linux
  - [x] With fully cleaned.
- [x] Check installation and working with Windows ( I will check soon. )
- [x] Check installation and working with Mac ( I can't! )

NOTE: `prisma-language-server` looks still buggy.

I welcome any suggestions. Thank you!

Filetype `prisma` is set by https://github.com/pantharshit00/vim-prisma .

### CODE REVIEW

File: settings.json [hunk 1]
```diff
@@ -601,6 +601,14 @@
...
     }
   ],
+  "prisma": [
+    {
+      "command": "prisma-language-server",
+      "requires": [
+        "npm"
+      ]
+    }
+  ],
```
Comment (heavenshell) [inline settings.json:611]: I think we may add vim-prisma to settings.json
```json
      "vim_plugin": {
        "extensions": [
          "prisma"
        ],
        "name": "pantharshit00/vim-prisma"
      }
```

Comment (LumaKernel) [reply settings.json]: Thank you! I added them.

**Comment by mattn** [comment]
Is this still WIP ?

**Comment by LumaKernel** [comment]
Thank you. It's ready for review.

**Comment by heavenshell** [comment]
In my mac, prisma-language-server did not work well.
```
Thu Dec  3 00:26:19 2020:["<---",1,"prisma-language-server",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"(node:14681) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'prismaFmtBinPath' of null\n    at /Users/heavenshell/.local/share/vim-lsp-settings/servers/prisma-language-server/node_modules/@prisma/language-server/dist/src/server.js:172:67\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/heavenshell/.local/share/vim-lsp-settings/servers/prisma-language-server/node_modules/@prisma/language-server/dist/src/server.js:24:58)\n(Use `node --trace-warnings ...` to show where the warning was created)","type":1}}}]
Thu Dec  3 00:26:19 2020:["<---",1,"prisma-language-server",{"response":{"method":"window/logMessage","jsonrpc":"2.0","params":{"message":"(node:14681) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)","type":1}}}]
```

## Do prisma's quick start.
```
curl https://codeload.github.com/prisma/quickstart/tar.gz/master | tar -xz --strip=2 quickstart-master/typescript/starter
cd starter
npm i
```

Change script.ts and prisma/schema.prisma
```
cat prisma/schema.prisma
datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}

generator client {
  provider = "prisma-client-js"
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
  published Boolean @default(false)
  author    User?   @relation(fields: [authorId], references: [id])
  authorId  Int?
}

model User {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
  posts Post[]
}
npm run dev
> ts-node ./script.ts
script@ dev /Users/heavenshell/Downloads/starter
[
  { id: 1, email: 'sarah@prisma.io', name: 'Sarah' },
  { id: 2, email: 'maria@prisma.io', name: 'Maria' }
]
```
prisma works fine.

## setup vim-lsp and vim-prisma
```vim
:LspInstallServer prisma-language-server
```
Install `vim-prisma`.


Open prisma/schema.prisma and do completion then, did not complete anything.
Also, run `LspDocumentFormat` raise error
```
Failed to document format for prisma-language-server: Request textDocument/formatting failed with message: Cannot read property 'prismaFmtBinPath' of null
```

macOS Mojavi
Vim 8.2

Any ideas?

File: installer/install-prisma-language-server.cmd [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+
+call npm install "@prisma/engines"
+
+echo @echo off ^
+
+call %%~dp0\node_modules\@prisma\engines\prisma-fmt-windows.exe %%* ^
+
+> prisma-fmt.exe
```
Comment (mattn) [inline installer/install-prisma-language-server.cmd:16]: should be `prisma-fmt.cmd`.

**Comment by LumaKernel** [comment]
Thank you for your patience, the cause of error is the change of prisma-language-tools, and not specific for macOS.
https://github.com/mattn/vim-lsp-settings/pull/340/commits/4ae98500e7421ab44559259c92c6133a2d586ad9

**Comment by heavenshell** [comment]
LGTM!! works fine in my env!

**Review by mattn — APPROVED** [review]

**Comment by mattn** [comment]
Thank you

**Pull request merged**